lib/storage/file.js

```diff
@@ -277,6 +277,28 @@ function File(bucket, name, options) {
 nodeutil.inherits(File, ServiceObject);
 
 /**
+ * Generate a padded aes-256 base64 key for encryption.
+ *
+ * @param {string} code - A user-friendly passphrase.
+ * @return {string} - The base64 aes-256 key.
+ */
+File.encrypt_ = function(code) {
+  var cipher = crypto.createCipher('aes-256-ctr', 'gcloud-node');
+
+  var paddingLength = 32 - code.length;
+  while (paddingLength--) {
+    code += ' ';
+  }
+
+  var key = cipher.update(code, 'utf8', 'base64') + cipher.final('base64');
+
+  return {
+    key: key,
+    hash: crypto.createHash('sha256').update(key, 'base64').digest('base64')
+  };
+};
```

Comment:
When you upload a file, you should be able to provide a custom key. Is that doable here?